static_frame/core/frame.py

```diff
@@ -5968,6 +6021,55 @@ def to_latex(self,
                 )
         return repr(self.display(config))
 
+    def to_msgpack(self) -> 'bin':
+        '''
+        Return a msgpack.
+        '''
+        import msgpack
+        import msgpack_numpy
+        def encode(obj, chain=msgpack_numpy.encode):
+            clsname = obj.__class__.__name__
+            package = obj.__class__.__module__.split('.',1)[0]
+            print('package, clsname', package, clsname)
+            if package == 'static_frame':
+                if clsname in ['Index', 'IndexBase']:
+                    return {b'sf':clsname,
+                            b'name':obj.name,
+                            b'data':packb(obj.values)} #recursively call packb to nest msgpacks within
+                elif clsname in ['IndexHierarchy', 'IndexHierarchyGO']:
+                    if obj._recache:
+                        obj._update_array_cache()
+                    return {b'sf':clsname,
+                            b'name':obj.name,
+                            b'data':packb(obj._blocks._blocks), #recursively call packb to nest msgpacks within
+                            b'levels':packb(obj._levels)} #recursively call packb to nest msgpacks within
+                elif clsname in ['IndexLevel']:
+                    return {b'sf':clsname,
+                            b'index':packb(obj.index), #recursively call packb to nest msgpacks within
+                            b'_depth':obj._depth,
+                            b'_length':obj._length}
+                elif clsname in ['Frame']:
+                    return {b'sf':clsname,
+                            b'name':obj.name,
+                            b'data':packb(obj._blocks._blocks), #recursively call packb to nest msgpacks within
+                            b'index':packb(obj.index), #recursively call packb to nest msgpacks within
+                            b'columns':packb(obj.columns)} #recursively call packb to nest msgpacks within
+                elif clsname in ['IndexDate']:
+                    return {b'sf':clsname,
+                           }
+            if package == 'numpy':
+                if isinstance(obj, np.ndarray):
+                    if isinstance(obj[0], np.datetime64) or isinstance(obj[0], np.timedelta64):
```

Comment:
Rather than looking at the type of the element in the array, you can simply look at the `kind` of the `dtype` attr, 
```
>>> np.array(('2020', '20201'), dtype=np.datetime64).dtype.kind
'M'
```
You want to handle here kinds 'M' (datetime64) and 'm' (timedelta64)

Reply:
```
#if obj.dtype.kind in ['M', 'm']:
 if obj.dtype.type in [np.datetime64, np.timedelta64]:
```
I spent a bunch of time today playing with dtype to understand it better. I dtype.type makes the code more clear than dtype.kind

Reply:
That is a fine approach. I have typically used dtype.kind because the string comparisons are fast and light-weight, but comparing the type is fine.